docs/source/plugins.rst

```diff
@@ -0,0 +1,194 @@
+.. _plugins:
+
+==============
+MLflow Plugins
+==============
+
+As a framework-agnostic tool for machine learning, the MLflow Python API provides developer APIs for
+writing plugins that integrate with different ML frameworks and backends.
+
+Plugins provide a powerful mechanism for customizing the behavior of the MLflow
+Python client, allowing you to integrate third-party tracking and artifact storage solutions,
+set special context tags at run creation, and override model registry methods.
+
+The MLflow Python API currently supports several types of plugins:
+
+* Tracking AbstractStore plugins: specify custom client behavior when users call
+  tracking API methods like :py:func:`mlflow.start_run`, :py:func:`mlflow.log_metric`, :py:func:`mlflow.log_param`.
+* ArtifactRepository plugins: specify custom client behavior when users call
+  :py:func:`mlflow.log_artifact`, :py:func:`mlflow.log_artifacts`
```

Comment:
Good call, agreed that it's important to emphasize the third-party storage use case. Ended up just adding a list of relevant plugin use cases (mentioning the third-party storage use case first) at the top of the plugins doc, so that people have a clear idea of what exactly plugins can be used (& are already used) to accomplish